src/backend/vulkan/src/device.rs

```diff
@@ -2234,6 +2235,82 @@ impl d::Device<B> for Device {
             _ => unreachable!(),
         }
     }
+
+    unsafe fn set_image_name(&self, image: &n::Image, name: &str) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::IMAGE, image.raw.as_raw(), name)
+    }
+
+    unsafe fn set_buffer_name(
+        &self,
+        buffer: &n::Buffer,
+        name: &str
+    ) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::BUFFER, buffer.raw.as_raw(), name)
+    }
+
+    unsafe fn set_command_buffer_name(
+        &self,
+        command_buffer: &cmd::CommandBuffer,
+        name: &str
+    ) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::COMMAND_BUFFER, command_buffer.raw.as_raw(), name)
+    }
+
+    unsafe fn set_semaphore_name(
+        &self,
+        semaphore: &n::Semaphore,
+        name: &str
+    ) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::SEMAPHORE, semaphore.0.as_raw(), name)
+    }
+
+    unsafe fn set_fence_name(&self, fence: &n::Fence, name: &str) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::FENCE, fence.0.as_raw(), name)
+    }
+
+    unsafe fn set_framebuffer_name(
+        &self,
+        framebuffer: &n::Framebuffer,
+        name: &str
+    ) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::FRAMEBUFFER, framebuffer.raw.as_raw(), name)
+    }
+
+    unsafe fn set_render_pass_name(
+        &self,
+        render_pass: &n::RenderPass,
+        name: &str
+    ) -> Result<(), d::OutOfMemory> {
+        self.set_object_name(vk::ObjectType::RENDER_PASS, render_pass.raw.as_raw(), name)
+    }
+}
+
+impl Device {
+    unsafe fn set_object_name(&self, object_type: vk::ObjectType, object_handle: u64, name: &str)
+    -> Result<(), d::OutOfMemory> {
+        let instance = &self.raw.2;
+        if let Some(ref debug_utils_ext) = instance.1 {
+            let name = CString::new(name).unwrap();
```

Comment:
let's leave a TODO here to avoid heap allocating this string

Reply:
Vulkan requires a null-terminated string here, but Rust strings are not null-terminated so I'm not sure how to avoid this allocation

Reply:
The way to do this would be to have some kind of temporary storage associated with `Device`, which we'd use to copy the string in, append a null terminator, and provide to Vulkan then.